mlflow/models/model.py

```diff
@@ -12,6 +12,7 @@
 
 import yaml
 from packaging.requirements import InvalidRequirement, Requirement
+from typing_extensions import Self
```

Comment:
does mlflow have typing_extensions as a dependency?

Reply:
Based on the Python 3.8 pip environment I got when running `dev/dev-env-setup.sh -d .venvs/mlflow-dev -q` from the contributions guide + querying with `pipdeptree`: `pipdeptree --reverse --packages typing_extensions`
 it does not seem to be a _explicit_ dependency in core-requirements.txt, but it is a direct dependency of both `alembic` and `graphene`, that is listed.

Reply:
We should not use a package that mlflow doesn't depend on directly. Can we just a forward reference?

Reply:
It seems we can, wasn't aware of that syntax. Thanks for the tip!